### PR TITLE
upload missing keys probably sick

### PR DESCRIPTION
**Pull request by bastianh**

### CODE REVIEW

File: CoronaContact/Scenes/Revocation/StatusReport/RevocationStatusReportViewModel.swift [hunk 1]
```diff
@@ -42,8 +42,12 @@ class RevocationStatusReportViewModel: ViewModel {
         guard isValid else {
             return
         }
+        guard let isProbablySickAt = localStorage.isProbablySickAt else { fatalError() }
 
-        flowController.submit { [weak self] result in
+        let startDate = isProbablySickAt.addDays(-3)!
```
Comment (finnsch) [inline CoronaContact/Scenes/Revocation/StatusReport/RevocationStatusReportViewModel.swift]: Can you specify this number somewhere in a Configuration type?

File: CoronaContact/Scenes/SelfTesting/Report/StatusReport/SelfTestingStatusReportViewModel.swift [hunk 2]
```diff
@@ -9,43 +9,60 @@ import Resolver
...
             return
         }
 
-        flowController.submit { [weak self] result in
-            completion()
+        var startDate = Date().addDays(-3)!
+        var endDate = Date()
 
+        if updateKeys {
+            if let missingUploadedKeysAt = localStorage.missingUploadedKeysAt {
```
Comment (finnsch) [inline CoronaContact/Scenes/SelfTesting/Report/StatusReport/SelfTestingStatusReportViewModel.swift]:
```suggestion
        if updateKeys, let missingUploadedKeysAt = localStorage.missingUploadedKeysAt {
```

File: CoronaContact/Scenes/SicknessCertificate/StatusReport/SicknessCertificateStatusReportViewModel.swift [hunk 3]
```diff
@@ -29,25 +30,43 @@ class SicknessCertificateStatusReportViewModel: ViewModel {
...
 
-        flowController.submit { [weak self] result in
+        var startDate = Date().addDays(-3)!
+        if let isProbablySickAt = localStorage.isProbablySickAt {
+            startDate = isProbablySickAt.addDays(-3)!
+        }
+        var endDate = Date()
+
+        if updateKeys {
+            if let missingUploadedKeysAt = localStorage.missingUploadedKeysAt {
```
Comment (finnsch) [inline CoronaContact/Scenes/SicknessCertificate/StatusReport/SicknessCertificateStatusReportViewModel.swift]:
```suggestion
        if updateKeys, let missingUploadedKeysAt = localStorage.missingUploadedKeysAt {
```

File: CoronaContact/Services/Exposure/ExposureKeyManager.swift [hunk 4]
```diff
@@ -3,17 +3,60 @@
...
+                    }
+
+                    let timestamps = filteredKeys.map(\.rollingStartNumber)
+                    let passwords = try TracingKeyPassword.getPasswordsFor(timestamps: timestamps)
+
+                    let temporaryExposureKeys = filteredKeys.map { key in
+                        TemporaryExposureKey(temporaryExposureKey: key, password: passwords[key.rollingStartNumber])
+                    }
+                    #if DEBUG
+                        LoggingService.debug("---\nUploading Keys:")
```
Comment (finnsch) [inline CoronaContact/Services/Exposure/ExposureKeyManager.swift]: Can you add a context here?

**Review by finnsch — CHANGES_REQUESTED** [review]

File: CoronaContact/Scenes/Revocation/StatusReport/RevocationStatusReportViewModel.swift (hunk 1 above)
Comment (bastianh) [reply CoronaContact/Scenes/Revocation/StatusReport/RevocationStatusReportViewModel.swift]: I think this should go into the configuration.json

**Review by finnsch — APPROVED** [review]

**Pull request merged**